### PR TITLE
Rexi updated

### PR DESCRIPTION
**Pull request by jshipton**
This contains code to compute the exponential of the linear operator defined by linearising an equation.

The code is tested for shallow water and compressible Boussinesq and there are also some tests for the REXI approximation.

### CODE REVIEW

**Review by jshipton — COMMENTED** [review]
@nhartney could you update the checkpoint data file to contain known correct output from REXI so that we can then make the test tolerances smaller? Thanks!

File: gusto/core/io.py [hunk 1]
```diff
@@ -21,7 +21,7 @@ class GustoIOError(IOError):
     pass
 
 
-def pick_up_mesh(output, mesh_name):
+def pick_up_mesh(output, mesh_name, comm=COMM_WORLD):
```
Comment (tommbendall) [inline gusto/core/io.py:24]: We had a brief chat about the addition of the `comm` argument here last week -- this makes sense to me now.

**Review by tommbendall — APPROVED** [review]
I am happy with everything here -- we just need to merge in main and resolve conflicts

**Review by tommbendall — APPROVED** [review]
Thanks Jemma, exciting!

**Pull request merged**